test/TestAsset/WhereInvokable.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace ZendTest\Db\TestAsset;
+
+use Zend\Db\Sql\Predicate\Like;
+use Zend\Db\Sql\Select;
+
+class WhereInvokable
+{
+    private $value;
+
+    /**
+     * WhereInvokable constructor.
+     * @param $value
+     */
+    public function __construct($value)
+    {
+        $this->value = $value;
+    }
+
+    public function __invoke($select)
```

Comment:
Add a type declaration to `$select`

Reply:
@I3ekka I've not read the code but I usually solve 'multi-type' by introducing an interface(with or without methods). If this commends totally misses the point, ignore it :P